src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1287 @@
+                        required to create empty raster array"""
+                    )
+
+                # create empty array,
+                dst_profile = {
+                    "count": band_count,
+                    "dtype": dtype,
+                    "driver": "TileDB",
+                    "crs": crs,
+                    "transform": from_origin(extents.minx, extents.maxy, *res),
+                    "width": math.floor((extents.maxx - extents.minx) / res[0]),
+                    "height": math.floor((extents.maxy - extents.miny) / res[1]),
+                    "blockysize": tile_size,
+                    "blockxsize": tile_size,
+                }
+
+                logger.debug("Destination raster profile %r", dst_profile)
+
+                if nodata is not None:
+                    dst_profile.update(
+                        {
+                            "nodata": nodata,
+                        }
+                    )
+
+                if compressor:
+                    compressor_args = dict(compressor)
+                    compressor_name = compressor_args.pop("_name")
+                    if compressor_name:
+                        compressor_args = {
+                            k: None if not v else v for k, v in compressor_args.items()
+                        }
+                        compression_filter = vars(filter).get(compressor_name)(
+                            **compressor_args
+                        )
+                        compression_level = compression_filter.level
+
+                    else:
+                        raise ValueError
+
+                    with rasterio.open(
+                        dataset_uri,
+                        mode="w",
+                        COMPRESSION=compressor_name[
+                            :-6
+                        ].upper(),  # remove trailing `Filter`
+                        COMPRESSION_LEVEL=compression_level,
+                        **dst_profile,
+                    ):
+                        pass
+                else:
+                    with rasterio.open(
+                        dataset_uri,
+                        mode="w",
+                        **dst_profile,
+                    ):
+                        pass
+
+                logger.debug("Raster array created %r", dataset_uri)
+            else:
+                # srcs and dst have same number of bands
+                if resampling:
+                    resampling = rasterio.enums.Resampling[resampling.lower()]
+                with rasterio.open(dataset_uri, mode="r+") as dst:
+                    for chunk in chunks:
+                        input_datasets = [
+                            rasterio.open(f, opener=vfs.open) for f in chunk.files
+                        ]
+                        try:
+                            row_off = chunk.ranges[0][0]
+                            col_off = chunk.ranges[1][0]
```

Comment:
This should probably be rearranged to `col_off` first to match below. (Alternately, pass in the params to `Window` as named arguments.)

Reply:
👍